JASP-Desktop/gui/preferencesmodel.cpp

```diff
@@ -458,3 +461,78 @@ void PreferencesModel::resetRememberedModules(bool setToRemember)
 {
 	setModulesRemembered(!setToRemember ? QStringList({}) : RibbonModel::singleton()->getModulesEnabled());
 }
+
+winLcCtypeSetting _getWinLcCtypeSetting()
+{
+	QString lcCtypeSetting = Settings::value(Settings::LC_CTYPE_C_WIN).toString();
+	
+	winLcCtypeSetting val = winLcCtypeSetting::check;
+	
+	try
+	{
+		val = winLcCtypeSettingFromQString(lcCtypeSetting);	
+	}
+	catch(missingEnumVal & e) {} //Just keep it at check then
+	
+	return val;
+}
+
+bool PreferencesModel::lcCtypeCheck() const
+{
+	return _getWinLcCtypeSetting() == winLcCtypeSetting::check;
+}
+
+bool PreferencesModel::lcCtypeToC() const
+{
+	return _getWinLcCtypeSetting() == winLcCtypeSetting::alwaysC;
+}
+
+void PreferencesModel::setLcCtypeToC(bool newVal)
+{
+	winLcCtypeSetting	current = _getWinLcCtypeSetting(),
+						newSet	= newVal ? winLcCtypeSetting::alwaysC : winLcCtypeSetting::neverC; 
+	
+	
+	if(current != newSet)
+	{
+		Settings::setValue(Settings::LC_CTYPE_C_WIN, winLcCtypeSettingToQString(newSet));
+		
+		emit lcCtypeToCChanged();
+		emit lcCtypeCheckChanged();
```

Comment:
Yeah it is if it is `true` it gets set to false because the underlying enum gets a different value.
And a little extra update is ok I think. Smaller chance of things going wrong because an old value gets stuck somewhere.